src/AspNet.Security.OAuth.KakaoTalk/KakaoTalkAuthenticationOptions.cs

```diff
@@ -0,0 +1,57 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Security.Claims;
+using System.Text.Json;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.KakaoTalk.KakaoTalkAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.KakaoTalk
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="KakaoTalkAuthenticationHandler"/>.
+    /// </summary>
+    public class KakaoTalkAuthenticationOptions : OAuthOptions
+    {
+        public KakaoTalkAuthenticationOptions()
+        {
+            ClaimsIssuer = KakaoTalkAuthenticationDefaults.Issuer;
+            CallbackPath = KakaoTalkAuthenticationDefaults.CallbackPath;
+
+            AuthorizationEndpoint = KakaoTalkAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = KakaoTalkAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = KakaoTalkAuthenticationDefaults.UserInformationEndpoint;
+
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "id");
+            ClaimActions.MapCustomJson(ClaimTypes.Name, user => GetValue(user, "kakao_account:profile:nickname"));
+            ClaimActions.MapCustomJson(ClaimTypes.Email, user => GetValue(user, "kakao_account:email"));
+            ClaimActions.MapCustomJson(ClaimTypes.DateOfBirth, user => GetValue(user, "kakao_account:birthday"));
+            ClaimActions.MapCustomJson(ClaimTypes.Gender, user => GetValue(user, "kakao_account:gender"));
+            ClaimActions.MapCustomJson(ClaimTypes.MobilePhone, user => GetValue(user, "kakao_account:phone_number"));
+            ClaimActions.MapCustomJson(Claims.AgeRange, user => GetValue(user, "kakao_account:age_range"));
+            ClaimActions.MapCustomJson(Claims.YearOfBirth, user => GetValue(user, "kakao_account:birthyear"));
```

Comment:
These can all be simplified to use `ClaimActions.MapJsonSubKey(string, string, string)`.

Reply:
Modified.